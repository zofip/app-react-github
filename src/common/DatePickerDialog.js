import "date-fns";
import React, { useState } from "react";
import DateFnsUtils from "@date-io/date-fns";
import {
  MuiPickersUtilsProvider,
  KeyboardDatePicker
} from "@material-ui/pickers";

import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles(theme => ({
  root: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 350
  }
}));

const DatePickerDialog = ({
  label,
  typeInput,
  handleChange
}) => {
  const classes = useStyles();
  const [selectedDate, setSelectedDate] = useState(null);

  function handleDateChange(date) {
    setSelectedDate(date);
    handleChange(typeInput, date);
  }

  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <KeyboardDatePicker
        margin="normal"
        className={classes.root}
        id="date-picker-dialog"
        label={label}
        format="MM/dd/yyyy"
        value={selectedDate}
        onChange={handleDateChange}
        KeyboardButtonProps={{
          "aria-label": "change date"
        }}
      />
    </MuiPickersUtilsProvider>
  );
};

export default DatePickerDialog;
