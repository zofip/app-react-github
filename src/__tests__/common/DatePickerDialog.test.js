import React from 'react';
import { shallow } from 'enzyme';
import DatePickerDialog from '../../common/DatePickerDialog';

describe('<DatePickerDialog />', () => {
   it('Rendering', () => {
      const datePickerDialog = shallow(<DatePickerDialog />);
      expect(datePickerDialog).toMatchSnapshot();
   });
});