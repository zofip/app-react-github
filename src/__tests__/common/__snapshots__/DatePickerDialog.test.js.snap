// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePickerDialog /> Rendering 1`] = `
<MuiPickersUtilsProvider
  utils={[Function]}
>
  <PickerWithState
    KeyboardButtonProps={
      Object {
        "aria-label": "change date",
      }
    }
    allowKeyboardControl={true}
    className="makeStyles-root-1"
    format="MM/dd/yyyy"
    id="date-picker-dialog"
    invalidDateMessage="Invalid Date Format"
    margin="normal"
    maxDate={2100-01-01T00:00:00.000Z}
    maxDateMessage="Date should not be after maximal date"
    minDate={1900-01-01T00:00:00.000Z}
    minDateMessage="Date should not be before minimal date"
    onChange={[Function]}
    openTo="date"
    value={null}
    views={
      Array [
        "year",
        "date",
      ]
    }
  />
</MuiPickersUtilsProvider>
`;
